Sources/StreamChatUI/ChatMessageList/RETHINK/MessageContentView.swift

```diff
@@ -0,0 +1,501 @@
+        super.updateContent()
+
+        // Text
+        textView?.text = content?.text
+        
+        // Avatar
+        let placeholder = uiConfig.images.userAvatarPlaceholder1
+        if let imageURL = content?.author.imageURL {
+            authorAvatarView?.imageView.loadImage(from: imageURL, placeholder: placeholder)
+        } else {
+            authorAvatarView?.imageView.image = placeholder
+        }
+        
+        // Bubble view
+        if content?.type == .ephemeral {
+            bubbleView?.backgroundColor = uiConfig.colorPalette.popoverBackground
+        } else {
+            bubbleView?.backgroundColor = content?.isSentByCurrentUser == true ?
+                uiConfig.colorPalette.background2 :
+                uiConfig.colorPalette.popoverBackground
+        }
+
+        // Metadata
+        metadataView?.content = content.map {
+            .init(message: $0, isAuthorNameShown: layoutOptions?.contains(.authorName) == true)
+        }
+
+        // Quoted message view
+        quotedMessageView?.content = content?.quotedMessage.map {
+            .init(message: $0, avatarAlignment: $0.isSentByCurrentUser ? .right : .left)
+        }
+
+        // Thread info
+        if let replyCount = content?.replyCount {
+            threadReplyCountButton?.setTitle(L10n.Message.Threads.count(replyCount), for: .normal)
+        } else {
+            threadReplyCountButton?.setTitle(L10n.Message.Threads.reply, for: .normal)
+        }
+        let latestReplyAuthorAvatar = content?.latestReplies.first?.author.imageURL
+        threadAvatarView?.imageView.loadImage(from: latestReplyAuthorAvatar)
+
+        // Reactions view
+        reactionsBubbleView?.tailDirection = content
+            .map { $0.isSentByCurrentUser ? .toTrailing : .toLeading }
+        reactionsView?.content = content.map {
+            .init(
+                useBigIcons: false,
+                reactions: $0.reactions,
+                didTapOnReaction: nil
+            )
+        }
+    }
+
+    /// Cleans up the view so it is ready to display another message.
+    func prepareForReuse() {
+        content = nil
+        delegate = nil
+        indexPath = nil
+    }
+
+    // MARK: - Actions
+
+    /// Handles tap on `errorIndicatorView` and forwards the action to the delegate.
+    @objc open func handleTapOnErrorIndicator() {
+        guard let indexPath = indexPath else { return }
+
+        delegate?.didTapOnErrorIndicator(at: indexPath)
+    }
+
+    /// Handles tap on `threadReplyCountButton` and forwards the action to the delegate.
+    @objc open func handleTapOnThread() {
+        guard let indexPath = indexPath else { return }
+
+        delegate?.didTapOnThread(at: indexPath)
+    }
+
+    /// Handles tap on `quotedMessageView` and forwards the action to the delegate.
+    @objc open func handleTapOnQuotedMessage() {
+        guard let indexPath = indexPath else { return }
+
+        delegate?.didTapOnQuotedMessage(at: indexPath)
```

Comment:
I think we talked about this. The `indexPath` should be optional in the delegate callbacks. You can use this view without outside of the context of a cell

Reply:
✅ Done!